src/checkfiles/checkfiles.py

```diff
@@ -0,0 +1,197 @@
+
+    errors = {}
+    results = {}
+
+    is_gzipped = is_file_gzipped(file_path)
+    logging.info(f'is file gziped: {is_gzipped}')
+    check_valid_gzipped_file_format(errors, is_gzipped, file_format)
+    results['file_size'] = response.get('ContentLength')
+    check_file_size(errors, file_size, results['file_size'])
+    check_md5sum(errors, md5sum, response.get('ETag').strip('"'), file_path)
+
+    if is_gzipped:
+        check_content_md5sum(errors, file_path)
+
+    if file_format == 'bam':
+        bam_pysam_check(errors, file_path, number_of_reads)
+        if 'bam_error' not in errors:
+            bam_generate_bai_file(file_path)
+    elif file_format == 'fastq':
+        fastq_check(errors, file_path, number_of_reads, read_length)
+
+    logging.info(f'Completed file validation for file uuid {uuid}.')
+
+    if errors:
+        return {
+            'uuid': uuid,
+            'validation_result': 'failed',
+            'errors': errors
+        }
+    else:
+        return {
+            'uuid': uuid,
+            'validation_result': 'pass'
+        }
+
+
+def get_local_file_path(blob_name):
+    file_path = DATA_DIR + blob_name
+    return file_path
+
+
+def is_file_gzipped(file_path):
+    try:
+        gzip.GzipFile(filename=file_path).read(1)
+        return True
+    except gzip.BadGzipFile:
+        return False
+
+
+def check_valid_gzipped_file_format(errors, is_gzipped, file_format):
+    if file_format in ZIP_FILE_FORMAT and not is_gzipped:
+        errors['gzip'] = f'{file_format} file should be gzipped'
+    elif file_format not in ZIP_FILE_FORMAT and is_gzipped:
+        errors['gzip'] = f'{file_format} file should not be gzipped'
+
+
+def check_file_size(errors, file_size, size_in_cloud_storage):
+    if size_in_cloud_storage != file_size:
+        errors['file_size'] = f'submitted file zise {str(file_size)} does not mactch file zise {str(size_in_cloud_storage)} in cloud storage'
+
+
+def check_md5sum(errors, md5sum, etag, file_path, chunk_size=CHUNK_SIZE):
+    logging.info(f'the eTag is {etag}')
+    if etag != md5sum:
+        md5 = hashlib.md5()
+        with open(file_path, 'rb') as local_file:
+            while chunk := local_file.read(chunk_size):
+                md5.update(chunk)
+        calculated_md5sum = md5.hexdigest()
+
+        if md5sum != calculated_md5sum:
+            errors['md5sum'] = f'submitted file md5sum {(md5sum)} does not mactch file md5sum {calculated_md5sum} in cloud storage'
+
+
+def check_content_md5sum(errors, file_path, chunk_size=128*6400000):
+    md5 = hashlib.md5()
+    with gzip.open(file_path) as f:
+        while chunk := f.read(chunk_size):
+            md5.update(chunk)
+    content_md5sum = md5.hexdigest()
+    logging.info(f'content md5sum is {content_md5sum}')
+    url = CONTENT_MD5SUM_URL + content_md5sum
+    session = requests.Session()
+    username = os.getenv('ENCODE_ACCESS_KEY')
+    password = os.getenv('ENCODE_CECRET_KEY')
```

Comment:
Should not depend on the environment (either pass as parameter or figure some other way to get these).